test/runtime/runtime_test.go

```diff
@@ -25,37 +25,66 @@ func (s *testingRuntimeSuite) TestPrependEnvsPrefix(c *C) {
 		comment     string
 		cmd         string
 		env         map[string]string
+		soft        bool
 		expectedCmd string
 		err         string
 	}{
 		{
 			"no variable in environment",
 			"/node server.js",
 			map[string]string{},
+			false,
 			"/node server.js",
 			"",
 		},
 		{
 			"single variable in environment",
 			"/node server.js",
 			map[string]string{"PORT": "8000"},
+			false,
 			"--env=PORT=8000 /node server.js",
 			"",
 		},
 		{
 			"two variables in environment",
 			"/node server.js",
 			map[string]string{"PORT": "8000", "ENDPOINT": "foo.com"},
+			false,
 			// Order is not guaranteed hence the following regex is needed:
 			"(--env=PORT=8000 --env=ENDPOINT=foo.com|--env=ENDPOINT=foo.com --env=PORT=8000) /node server.js",
 			"",
 		},
+		{
+			"no variable in environment - soft",
+			"/node server.js",
+			map[string]string{},
+			true,
+			"/node server.js",
+			"",
+		},
+		{
+			"single variable in environment - soft",
+			"/node server.js",
+			map[string]string{"PORT": "8000"},
+			true,
+			"--env=PORT?=8000 /node server.js",
```

Comment:
Using `?` here will cause problems for your tests. As you can see [here](https://github.com/go-check/check/blob/v1/checkers.go#L287), the `c.Matches` that you are using below, will use regular expression. In terms of regex `?` stands for optional character in front of it, i.e. `PORT?` will match either `PORT` or `POR`. Since you are interested in `?=`, you have to escape `?`, i.e. `--env=PORT\?=8000...`.

All these tests are passing because you use `false` in L87, causing that all env settings are `hard`. Change L87 according to the comment, and these tests should start failing.